### PR TITLE
STYLE: Prefer C++11 type alias over typedef.

### PR DESCRIPTION
**Pull request by jhlegarreta**
Prefer C++11 type alias over typedef for the reasons stated here:
http://review.source.kitware.com/#/c/23103/

### CODE REVIEW

File: include/itkBSplineScatteredDataPointSetToGradientImageFilter.h [hunk 1]
```diff
@@ -62,46 +62,45 @@ class BSplineScatteredDataPointSetToGradientImageFilter :
...
+  using OutputValueType = TOutputValueType;
+  using OutputPixelType = CovariantVector< OutputValueType, ImageDimension >;
+  using OutputImageRegionType = typename OutputImageType::RegionType;
+  using InternalImageType = typename itk::Image< InputPixelType, ImageDimension >;
 
   /** Internal filter type. */
-  typedef typename itk::BSplineScatteredDataPointSetToImageFilter< InputPointSetType,
-                                                                   InternalImageType > BSplineScatteredDataFilterType;
-  typedef typename BSplineScatteredDataFilterType::Pointer BSplineScatteredDataFilterPointerType;
+  using BSplineScatteredDataFilterType = typename itk::BSplineScatteredDataPointSetToImageFilter< InputPointSetType,
```
Comment (dzenanz) [inline include/itkBSplineScatteredDataPointSetToGradientImageFilter.h]: Maybe break the line at `<`?

File: test/itkBSplineApproximationGradientImageFilterTest.cxx [hunk 2]
```diff
@@ -56,24 +56,23 @@ int itkBSplineApproximationGradientImageFilterTest( int argc, char* argv[] )
...
-    GradientImageFilterType;
+  using GradientImageFilterType = itk::BSplineApproximationGradientImageFilter< VectorImageType, float >;
   GradientImageFilterType::Pointer  gradientImageFilter = GradientImageFilterType::New();
   gradientImageFilter->SetInput( imageToVI->GetOutput() );
   gradientImageFilter->SetNumberOfLevels( 1 );
   gradientImageFilter->SetControlPointSpacingRatio( 2.0 );
 
-  typedef itk::VectorMagnitudeImageFilter< GradientImageFilterType::OutputImageType,
-                                               OutputImageType > GradientMagnitudeFilterType;
+  using GradientMagnitudeFilterType = itk::VectorMagnitudeImageFilter< GradientImageFilterType::OutputImageType,
```
Comment (dzenanz) [inline test/itkBSplineApproximationGradientImageFilterTest.cxx]: Break line at `<`?

File: test/itkBSplineScatteredDataPointSetToGradientImageFilterTest.cxx [hunk 3]
```diff
@@ -58,16 +58,15 @@ int itkBSplineScatteredDataPointSetToGradientImageFilterTest( int argc, char *ar
...
 
-  typedef itk::ImageToPointSetFilter< VectorImageType, MeshType >
-    ImageToPointSetType;
+  using ImageToPointSetType = itk::ImageToPointSetFilter< VectorImageType, MeshType >;
   ImageToPointSetType::Pointer imageToPointSet = ImageToPointSetType::New();
   imageToPointSet->SetInput( 0, imageToVI->GetOutput() );
 
-  typedef itk::BSplineScatteredDataPointSetToGradientImageFilter< MeshType,
-                                                                  PixelType >  PointSetToGradientFilterType;
+  using PointSetToGradientFilterType = itk::BSplineScatteredDataPointSetToGradientImageFilter< MeshType,
```
Comment (dzenanz) [inline test/itkBSplineScatteredDataPointSetToGradientImageFilterTest.cxx]: Break line at `<`, `=` or have no line break?

File: test/itkBSplineScatteredDataPointSetToGradientImageFilterTest.cxx [hunk 4]
```diff
@@ -78,12 +77,12 @@ int itkBSplineScatteredDataPointSetToGradientImageFilterTest( int argc, char *ar
   pointSetToGradient->SetNumberOfControlPoints( ncps );
   pointSetToGradient->SetNumberOfLevels( 3 );
 
-  typedef itk::VectorMagnitudeImageFilter< PointSetToGradientFilterType::OutputImageType,
-                                               OutputImageType > GradientMagnitudeFilterType;
+  using GradientMagnitudeFilterType = itk::VectorMagnitudeImageFilter< PointSetToGradientFilterType::OutputImageType,
```
Comment (dzenanz) [inline test/itkBSplineScatteredDataPointSetToGradientImageFilterTest.cxx]: Break line at `<`.

**Review by dzenanz — COMMENTED** [review]
Thanks for working on this. Minor suggestions in-line.

File: test/itkBSplineScatteredDataPointSetToGradientImageFilterTest.cxx (hunk 4 above)
Comment (jhlegarreta) [reply test/itkBSplineScatteredDataPointSetToGradientImageFilterTest.cxx]: Done in c577d13.

File: test/itkBSplineScatteredDataPointSetToGradientImageFilterTest.cxx (hunk 3 above)
Comment (jhlegarreta) [reply test/itkBSplineScatteredDataPointSetToGradientImageFilterTest.cxx]: Done in c577d13.

File: test/itkBSplineApproximationGradientImageFilterTest.cxx (hunk 2 above)
Comment (jhlegarreta) [reply test/itkBSplineApproximationGradientImageFilterTest.cxx]: Done in c577d13.

File: include/itkBSplineScatteredDataPointSetToGradientImageFilter.h (hunk 1 above)
Comment (jhlegarreta) [reply include/itkBSplineScatteredDataPointSetToGradientImageFilter.h]: Still found it too long, although I know this is subjective. Preferred to break at `=` in c577d13.

**Pull request merged**